### PR TITLE
Fixed #594 - Wrong service scope  in web components

### LINKED ISSUE #594: Can't implement registered custom Handlebars helpers anywhere.
!! Check existing issues before logging new ones !!

**Version used**
feature/v4

**Describe the bug**
search-extensibility-demo has a default custom Hundlebar helper called 'myHelper'. When I try to call it on some values in searchResults I get the message "Error: Missing helper: "myHelper".

**To Reproduce**
1.   Clone feature/v4 branch
2.   Navigate to folder search-extensibility
3.   run npm i
4.   run gulp bundle
5.   run npm link
6.   Navigate to folder search-extensibility-demo
7.   run npm i
8.   run npm link @pnp/modern-search-extensibility
9.   run gulp bundle
10. run gulp bundle --ship
11. run gulp package --ship
12. add the solution to the global catalog
13. select the checkbox "Make this solution available to all sites in the organization"
14. register my manifest ID in the target Web Part instance
15. add a column for the details list layout 
16. make adjustment to column value by implementing the default custom Handlebars helper "myHelper"  

**Expected behavior**
By definition "myHelper" should convert  a string-value to lowercase.

**Screenshots**

image
image
image
image

**Desktop (please complete the following information):**
 - Browser : chrome, edge

**Additional context**
none

### PR DESCRIPTION
**Pull request by FranckyC**
Service scope in web components wasn't the correct scope for the current displaying WebPart. Introduced an array of scope references to lookup the correct one inside a web component. Did the same for service keys as it is the same concern due to separate Web Part bundle.

### CODE REVIEW

**Comment by FranckyC** [comment]
@wobba Found the root cause of this issue. Sorry I removed your previous hack ;). I would need a new version of the extensibility npm module to be published as I introduced new properties.

**Comment by wobba** [comment]
Awesome! I'll publish a new npm np.

File: search-parts/package.json [hunk 1]
```diff
@@ -25,7 +25,6 @@
         "@microsoft/sp-property-pane": "1.12.1",
         "@microsoft/sp-webpart-base": "1.12.1",
         "@pnp/common": "2.0.13",
-        "@pnp/modern-search-extensibility": "1.0.2",
```
Comment (wobba) [inline search-parts/package.json:28]: Should this be kept and updated to 1.0.3 once I publish a new solution?

**Comment by wobba** [comment]
Merged, and published a new version of the extlib npm.

**Comment by FranckyC** [comment]
Thanks @wobba but I didn't update the package.json yet...Please don't merge my PRs until it's done, I can handle it ;)

**Comment by wobba** [comment]
I updated the npm with the change in this PR 1.0.3. If you need more changes, then comment on the next PR when it's ready.

**Comment by FranckyC** [comment]
Should be OK for this one

**Pull request merged**